### DISCUSSION TITLE
Autouse fixture for certain markers

### DISCUSSION THREAD

**Discussion by akvilary** [pytest-dev/pytest#12149] [Ideas]
Dears,
Set markers is so useful to differentiate and select tests after.
But it would be more useful if we could register fixtures for certain markers.
Scenario:
  - once register fixture for certain marker with autouse_for_markers argument.
  - every time later user only has to mark test or testing class to use all registered fixtures for the mark and differentiate tests (no need to use extra @pytest.mark.usefixtures decorator with custom mark).
 
Today it looks like:
```python
@pytest.fixture(autouse=True)
def create_transaction(request):
    """
    Fixture create transaction in db, and rollback after
    """
    if 'db' in request.keywords or 'another_marker' in request.keywords:
        # setup
        transaction = CreateTransaction(
            TransactionLevel.REPEATABLE_READ,
            TransactionMode.WRITE,
        )
        save_point_name = 'SAVE_POINT'
        transaction.CreateSavePoint(save_point_name)
        yield
        # teardown
        transaction.RollbackToSavePoint(save_point_name)
        transaction.Break()
    else:
        yield
```
 
My suggestion is:
```python
@pytest.fixture(autouse_for_markers=['db', 'another_marker'])
def create_transaction():
    """
    Fixture create transaction in db, and rollback after
    """
    # setup
    transaction = CreateTransaction(
        TransactionLevel.REPEATABLE_READ,
        TransactionMode.WRITE,
    )
    save_point_name = 'SAVE_POINT'
    transaction.CreateSavePoint(save_point_name)
    yield
    # teardown
    transaction.RollbackToSavePoint(save_point_name)
    transaction.Break()
```
So next time I have to only mark (fixtures will apply automatically for the mark):
```python
@pytest.mark.db
def test_simple_insert():
    ...
````

**Comment by RonnyPfannschmidt** [comment]
for this proposal a strict no 

at first glace dependencies and order are completely undefined

without more context this looks like a extra headache for magic globals

it would be much safer/saner to just accept a fixture that provides the db session to interact with in a meaningful way

**Reply by RonnyPfannschmidt** [reply]
it may be feasible to have autouse be a callable, however its unclear what to do with non-function scopes for conditional autouse
